packages/react-native/ReactCommon/react/renderer/components/view/ViewProps.cpp

```diff
@@ -412,6 +422,56 @@ BorderMetrics ViewProps::resolveBorderMetrics(
   };
 }
 
+Transform ViewProps::resolveTransform(
+  LayoutMetrics const &layoutMetrics) const  {
+  float viewWidth = layoutMetrics.frame.size.width;
+  float viewHeight = layoutMetrics.frame.size.height;
+  if (transformOrigin.empty() || (viewWidth == 0 && viewHeight == 0)) {
+    return transform;
+  }
+  auto newTransform = Transform{};
+  std::array<float, 3> translateOffsets = getTranslateForTransformOrigin(viewWidth, viewHeight);
+  newTransform = newTransform * Transform::Translate(translateOffsets[0], translateOffsets[1], translateOffsets[2]);
+  newTransform = newTransform * transform;
+  newTransform = newTransform * Transform::Translate(-translateOffsets[0], -translateOffsets[1], -translateOffsets[2]);
+  return newTransform;
+}
+
+// https://drafts.csswg.org/css-transforms/#transform-origin-property
+std::array<float, 3> ViewProps::getTranslateForTransformOrigin(float viewWidth, float viewHeight) const {
+  float viewCenterX = viewWidth / 2;
+  float viewCenterY = viewHeight / 2;
+
+  std::array<float, 3> origin = {viewCenterX, viewCenterY, 0.0f};
+  std::istringstream iss(transformOrigin);
+  std::string part;
+  for (int i = 0; std::getline(iss, part, ' ') && i < 3; i++) {
+    auto percentPos = part.find('%');
+    bool isPercent = percentPos != std::string::npos;
```

Comment:
`istringstream` and `getline` are pretty inefficient string parsing methods.

I'd use `std::find` instead to find the next space (or end), then use `string_view` (like [here](https://github.com/facebook/react-native/blob/f6f848f9da4d3618db61a544f854cbf54c7b577d/packages/react-native/ReactCommon/react/renderer/components/view/conversions.h#L422)) to parse it.

Reply:
Instead of parsing this on every transform calculation, consider parsing it as and storing it as a YGValue, using `YGUnitPercent` and`YGUnitPoint`.

Reply:
TODO: this still needs to move away from getline, and look at adopting `YGValue` to avoid parsing this on every layout change